modal/_utils/http_utils.py

```diff
@@ -25,6 +27,25 @@ def http_client_with_tls(timeout: Optional[float]) -> ClientSession:
     return ClientSession(connector=connector, timeout=ClientTimeout(total=timeout))
 
 
+class ClientSessionRegistry:
+    _client_session: ClientSession
+    _client_session_active: bool = False
+
+    @staticmethod
+    def get_session():
+        if not ClientSessionRegistry._client_session_active:
+            ClientSessionRegistry._client_session = _http_client_with_tls(timeout=None)
+            ClientSessionRegistry._client_session_active = True
+            on_shutdown(ClientSessionRegistry.close_session())
```

Comment:
Any chance we end up leaving the session open if e.g. we crash in `_http_client_with_tls`? Would that be a problem?

Reply:
@luiscape

Reply:
@luiscape

Reply:
We definitely have a lot of rough edges in our shutdown behavior so I wouldn't assume that what we currently have is right way to do it — but good to know at least it won't be a regression :)  BTW do we also need to toggle `_client_session_active` on shutdown too? Probably we can assume nothing else that runs at or after that point will try to get a session?

Reply:
We don't but I think it's good to make the close_session function general enough that it can be called before shutdown as well if necessary.